mllam_data_prep/config.py

```diff
@@ -151,13 +169,20 @@ class InputDataset:
         (e.g. two datasets that coincide in space and time will only differ in the feature dimension,
         so the two will be combined by concatenating along the feature dimension).
         If a single shared coordinate cannot be found then an exception will be raised.
+    derived_variables: Dict[str, DerivedVariable]
+        Dictionary of variables to derive from the dataset, where the keys are the variable names and
+        the values are dictionaries defining the necessary function and kwargs. E.g.
+        `{"toa_radiation": {"kwargs": {"time": "time", "lat": "lat", "lon": "lon"}, "function": "calculate_toa_radiation"}}`
+        would derive the "toa_radiation" variable using the `calculate_toa_radiation` function, which
+        takes `time`, `lat` and `lon` as arguments.
     """
 
     path: str
     dims: List[str]
-    variables: Union[List[str], Dict[str, Dict[str, ValueSelection]]]
     dim_mapping: Dict[str, DimMapping]
     target_output_variable: str
+    variables: Union[List[str], Dict[str, Dict[str, ValueSelection]]] = None
+    derived_variables: Dict[str, DerivedVariable] = None
     attributes: Dict[str, Any] = None
```

Comment:
I am not sure that I understand what isn't documented. Do you mean that `variables` and `derived_variables` are not documented in the docstring to `InputDataset` or something else? Because the docstring does include documentation of `variables` and `derived_variables` (lines 155-159 + 172-173; sorry not sure how to format this with the lines highlighted in a reply to a comment).

Reply:
No, it was just, there is no entry in the docstring for the "attributes" attribute ;)

Reply:
Ah, I see. This was added before this PR but I have made an attempt to document it in the docstring. Hopefully it is correct, @leifdenby ?

Reply:
> But in the case for derived variables (which would be a different dataclass, the `DerivedVariable` one I thought we had said that we wanted to provide the user a way to _set the attributes in the case where what is returned from a function doesn't have those required attributes_. Am I remembering that correctly?

No, you are correct. The `attributes` for the `DerivedVariable` class is a way to set attributes for the derived variables. Form what I remember, we discussed that it should do the following (which it also does at the moment):
- If the attributes (at the moment `units` and `long_name`) are not set by the function then they need to be specified in the config file through the `attributes` section of the `DerivedVariable` 
- If the attributes are set in the function and also defined in the config file then the attributes from the config file will overwrite the ones from the function and a warning will be issued.
- If the attributes are set in the function and not defined in the config file, nothing will happen. 
- If the attributes are not set in the function nor in the config file, we raise an error